internal/pkg/cli/run_local.go

```diff
@@ -256,6 +270,32 @@ func newRunLocalOpts(vars runLocalVars) (*runLocalOpts, error) {
 	o.newRecursiveWatcher = func() (recursiveWatcher, error) {
 		return file.NewRecursiveWatcher(0)
 	}
+
+	// Capture stdout by replacing it with a piped writer and returning an attached io.Reader.
+	// Functions are concurrency safe and idempotent.
+	var mu sync.Mutex
+	var savedWriter, savedStdout *os.File
+	savedStdout = os.Stdout
+	o.captureStdout = func() (io.Reader, error) {
+		if savedWriter != nil {
+			savedWriter.Close()
+		}
+		pipeReader, pipeWriter, err := os.Pipe()
+		if err != nil {
+			return nil, err
+		}
+		mu.Lock()
+		savedWriter = pipeWriter
+		os.Stdout = savedWriter
+		mu.Unlock()
```

Comment:
Should we use a defer statement to have lock and unlock next to each other
```suggestion
		mu.Lock()
		defer mu.Unlock()
		savedWriter = pipeWriter
		os.Stdout = savedWriter
```